ansible_collections/arista/avd/roles/eos_designs/python_modules/connected_endpoints/port_channel_interfaces.py

```diff
@@ -0,0 +1,172 @@
+                {
+                    "mode": adapter.get("mode"),
+                    "l2_mtu": adapter.get("l2_mtu"),
+                    "vlans": adapter.get("vlans"),
+                    "trunk_groups": self._get_adapter_trunk_groups(adapter, connected_endpoint),
+                    "native_vlan_tag": adapter.get("native_vlan_tag"),
+                    "native_vlan": adapter.get("native_vlan"),
+                    "spanning_tree_portfast": adapter.get("spanning_tree_portfast"),
+                    "spanning_tree_bpdufilter": adapter.get("spanning_tree_bpdufilter"),
+                    "spanning_tree_bpduguard": adapter.get("spanning_tree_bpduguard"),
+                    "storm_control": self._get_adapter_storm_control(adapter),
+                }
+            )
+
+        # EVPN A/A
+        if (short_esi := self._get_short_esi(adapter, channel_group_id)) is not None:
+            port_channel_interface["evpn_ethernet_segment"] = self._get_adapter_evpn_ethernet_segment_cfg(adapter, short_esi, node_index, connected_endpoint)
+            if port_channel_mode == "active":
+                port_channel_interface["lacp_id"] = generate_lacp_id(short_esi)
+
+        # Set MLAG ID on port-channel if connection is multi-homed and this switch is running MLAG
+        elif self._mlag and len(set(adapter["switches"])) > 1:
+            port_channel_interface["mlag"] = channel_group_id
+
+        # LACP Fallback
+        if port_channel_mode == "active" and (lacp_fallback_mode := get(adapter, "port_channel.lacp_fallback.mode")) == "static":
+            port_channel_interface.update(
+                {
+                    "lacp_fallback_mode": lacp_fallback_mode,
+                    "lacp_fallback_timeout": get(adapter, "port_channel.lacp_fallback.timeout", default=90),
+                }
+            )
+
+        return strip_null_from_data(port_channel_interface)
+
+    def _get_port_channel_subinterface_cfg(self, subinterface: dict, adapter: dict, channel_group_id: int) -> dict:
+        """
+        Return structured_config for one port_channel_interface (subinterface)
+        """
+        # Common port_channel_interface settings
+        port_channel_interface = {
+            "type": "l2dot1q",
+            "vlan_id": subinterface.get("vlan_id", subinterface["number"]),
+            "encapsulation_vlan": {
+                "client": {
+                    "dot1q": {
+                        "vlan": get(subinterface, "encapsulation_vlan.client_dot1q", default=subinterface["number"]),
+                    }
+                },
+                "network": {
+                    "client": True,
+                },
+            },
+        }
+
+        # EVPN A/A
+        if (
+            "short_esi" in subinterface
+            and (short_esi := self._get_short_esi(adapter, channel_group_id, short_esi=subinterface["short_esi"], hash_extra_value=subinterface["number"]))
+            is not None
+        ):
```

Comment:
```suggestion
            (short_esi := self._get_short_esi(adapter, channel_group_id, short_esi=subinterface.get("short_esi", None), hash_extra_value=subinterface["number"]))
            is not None
        ):
```

Reply:
This is not the same, as it will call the function and return. And it adds to the very long condition. I will see how to make it readable.

Reply:
done